sourir.ml

```diff
@@ -16,10 +16,14 @@ let () =
         exit 2
     in
     let quiet = Array.mem "--quiet" Sys.argv in
-    let prune = Array.mem "--prune" Sys.argv in
-    let codemotion = Array.mem "--cm" Sys.argv in
-    let constprop = Array.mem "--prop" Sys.argv in
-    let lifetime = Array.mem "--lifetime" Sys.argv in
+    let rec find_opts = function
+      | [] -> []
+      | o::r when String.length o > 5 && String.sub o 0 6 = "--opt=" ->
+        let opts = String.sub o 6 ((String.length o) - 6) in
+        String.split_on_char ',' opts
+      | o::r -> find_opts r
+    in
+    let optimizations = find_opts (Array.to_list Sys.argv) in
```

Comment:
I think it would be worth it to have a look at the documentation of the `Arg` module for how to actually do this properly. (I never had the time to squash that particular insect.)

(There are even fanciers/nicer way than Arg, in particular the [Cmdliner](http://erratique.ch/software/cmdliner) library which is extremely nice, but has a higher entry cost and I think that's not where we want to spend our sophistication right ow.)